pkg/server/local_observer.go

```diff
@@ -375,20 +375,20 @@ func (r *flowsReader) Next(ctx context.Context) (*pb.Flow, error) {
 		if e == nil {
 			return nil, io.EOF
 		}
+		if r.timeRange {
+			ts, err := types.TimestampFromProto(e.GetFlow().GetTime())
```

Comment:
Now that this has been moved before the type cast, `e.Getflow()` might return `<nil>`. So this code might potentially lead to a panic triggered by a nil pointer dereference.
What about assigning and checking the flow validity once just above after the `e == nil` check above?

Reply:
`v1.Event.GetFlow()` returns a typed `nil` and `pb.Flow.GetTime` is able to deal with the receiver being `nil`. `types.TimestampFromProto` also treats nil as the `zero` time, so it should not cause a panic. 

But I agree that introducing a nil check might be good practice either way however, in case any of the implementations change.

Reply:
Ha, fair enough. Let's leave that up to @tgraf, shouldn't block the PR then.